### PR TITLE
Add CircleCI configuration and automated publish process

### PR DESCRIPTION
**Pull request by jmonteiro**

### CODE REVIEW

**Comment by jmonteiro** [comment]
Tested by releasing a pre-release, confirming it worked: https://rubygems.org/gems/funneler/versions/1.2.2.pre1

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    steps:
+      - checkout
+      - run:
+          name: Install Bundler specific version
+          command: |
+            gem install bundler --version "${BUNDLE_VERSION}" --force
+      - restore_cache:
+          keys:
+            - v1-bundle-{{ checksum "Gemfile.lock" }}-
+      - run:
```
Comment (samusgray) [inline .circleci/config.yml]: Should we `bundle clean` here as well? From the Circle docs: https://circleci.com/docs/2.0/caching/#bundler-ruby

Comment (samusgray) [reply .circleci/config.yml]: Nope, ignore me. That's for partial dependency caching. Nothing to see here :D

Comment (samusgray) [reply .circleci/config.yml]: Maybe it's still a good idea?

**Pull request merged**